src/wx/faudio.cpp

```diff
@@ -194,43 +190,44 @@ class FAudio_Device_Notifier : public IMMNotificationClient {
 // Synchronization Event
 class FAudio_BufferNotify : public FAudioVoiceCallback {
 public:
-    HANDLE hBufferEndEvent;
+    void *hBufferEndEvent;
 
     FAudio_BufferNotify()
     {
         hBufferEndEvent = NULL;
         hBufferEndEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
         assert(hBufferEndEvent != NULL);
-    }
 
+        OnBufferEnd = &FAudio_BufferNotify::StaticOnBufferEnd;
+        OnVoiceProcessingPassStart = &FAudio_BufferNotify::StaticOnVoiceProcessingPassStart;
+        OnVoiceProcessingPassEnd = &FAudio_BufferNotify::StaticOnVoiceProcessingPassEnd;
+        OnStreamEnd = &FAudio_BufferNotify::StaticOnStreamEnd;
+        OnBufferStart = &FAudio_BufferNotify::StaticOnBufferStart;
+        OnLoopEnd = &FAudio_BufferNotify::StaticOnLoopEnd;
+        OnVoiceError = &FAudio_BufferNotify::StaticOnVoiceError;
```

Comment:
Since FAudioVoiceCallback is a struct in FAudio.h

couldn't I do something like

```
struct FAudio_BufferNotify {
  void OnBufferEnd(void* pBufferContext) {}
  FAudioVoiceCallback cb = {
   /* OnBufferEnd */ [](void* thiz, void* pBufferContext) { reinterpret_cast<FAudio_BufferNotify *>(thiz)->OnBufferEnd(pBufferContext); },
  // ... and the rest ...
  };
};
```

Reply:
Making the `FAudio_BufferNotify` the `FAudioVoiceCallback` object is a common pattern that allows you to more easily manage lifetimes. Basically, the idea is that the callback object always have to outlive the caller object. By making the callback object owned by the caller object, you have a strong guarantee from C++ semantics that the callback object is always going to outlive the caller object.

Reply:
Couldn't figure out how to do it differently, that will have to do for now.